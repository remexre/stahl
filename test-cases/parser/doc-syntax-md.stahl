foo
; This is equivalent to foo

foo bar
; This is equivalent to (foo bar)

foo bar baz
; This is equivalent to (foo bar baz)

foo bar
  baz
; This is equivalent to (foo bar baz)

foo
  bar
; This is equivalent to (foo bar)

foo
  bar baz
; This is equivalent to (foo (bar baz))

foo
  (bar baz)
; This is equivalent to (foo (bar baz))

foo
  bar
  baz
; This is equivalent to (foo bar baz)

(foo bar baz)
; This is equivalent to (foo bar baz)

(foo
  bar
  baz)
; This is equivalent to (foo bar baz)

foo
  bar
    baz
; This is equivalent to (foo (bar baz))

group
  foo
  bar
  baz
; This is equivalent to (foo bar baz)

group foo
  bar
  baz
; This is equivalent to (foo bar baz)

foo
  group 
    bar
    baz
; This is equivalent to (foo (bar baz))

foo
  bar
; The comment in the middle doesn't break the foo expression
  baz
  ; Neither does one indented here
  quux
    ; Or here!
  xyzzy
; This is equivalent to (foo bar baz quux xyzzy)

; Within an s-expression, indentation can be arbitrary:
(foo bar
     baz
  quux
    xyzzy)
; This is equivalent to (foo bar baz quux xyzzy)

; Indentation is tracked outside of the s-expression.
(foo bar
  baz)
  quux
; This is equivalent to ((foo bar baz) quux)

foo
  (bar
      baz
    quux)
  xyzzy
; This is equivalent to (foo (bar baz quux) xyzzy)
