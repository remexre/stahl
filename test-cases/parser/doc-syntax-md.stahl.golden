foo
(foo bar)
(foo bar baz)
(foo bar baz)
(foo bar)
(foo (bar baz))
(foo (bar baz))
(foo bar baz)
(foo bar baz)
(foo bar baz)
(foo (bar baz))
(foo bar baz)
(foo bar baz)
(foo (bar baz))
(foo bar baz quux xyzzy)
(foo bar baz quux xyzzy)
